### PR TITLE
Update pgo-build script to use the pinned Rust toolchain

### PR DESCRIPTION
**Pull request by caizixian**

### CODE REVIEW

**Comment by caizixian** [comment]
Actually, we might need to hold off this PR because PGO+LTO is broken on Rust 1.71.1

https://mmtk.zulipchat.com/#narrow/stream/262673-mmtk-core/topic/PGO/near/388531250
https://github.com/rust-lang/rust/issues/115344

**Comment by caizixian** [comment]
We should include PGO in the CI

File: .github/scripts/pgo-build.sh [hunk 1]
```diff
@@ -1,16 +1,24 @@
 #!/bin/bash
 
+pushd ../mmtk-openjdk/mmtk
+cargo clean
```
Comment (qinsoon) [inline .github/scripts/pgo-build.sh:4]: Why do we need `cargo clean`? Is it to work around the compiler bug? If so, probably add a comment so we may delete them once we move to a newer version.

**Review by qinsoon — APPROVED** [review]
LGTM

**Pull request merged**